### PR TITLE
Add message header forward for NATSToKafka connector

### PR DESCRIPTION
**Pull request by rpuppo**
Adding support for message header forward in nats-kafka connector in both directions.

### CODE REVIEW

**Comment by variadico** [comment]
Can you run `goimports -w .` at the top of the repo to fix the lints?
```
go install golang.org/x/tools/cmd/goimports@latest
```

And also `go vet ./...`.

File: server/core/connector.go [hunk 1]
```diff
@@ -216,6 +223,39 @@ func (conn *BridgeConnector) calculateKey(subject string, replyto string) []byte
...
 
+func (conn *BridgeConnector) convertFromKafkaToNatsHeaders(hdrs []sarama.RecordHeader) nats.Header {
+	// Iterate over all keys
+	if len(hdrs) > 0 {
+		nHdrs := make(nats.Header)
+		for i, kHdr := range hdrs {
+			if kHdr.Value != nil {
+				nHdrs.Add(string(kHdr.Key), string(kHdr.Value))
+			}
+			i++
```
Comment (variadico) [inline server/core/connector.go:234]: It doesn't seem like this `i` is being used for anything. Can we remove it?

File: server/kafka/consumer.go [hunk 2]
```diff
@@ -372,3 +376,12 @@ func (c *saramaConsumer) validateJSONSchema(schema *srclient.Schema, cleanPayloa
 
 	return cleanPayload, nil
 }
+
+func (c *saramaConsumer) convertToMessageHeaders(consumerHeaders []*sarama.RecordHeader) []sarama.RecordHeader {
+	var msgHeaders = make([]sarama.RecordHeader, len(consumerHeaders))
+	for i, element := range consumerHeaders {
+		msgHeaders[i] = *element
+		i++
```
Comment (variadico) [inline server/kafka/consumer.go:384]: It doesn't seem like this `i++` is doing anything. Can we remove it?

File: server/kafka/consumer_test.go [hunk 3]
```diff
@@ -114,6 +116,38 @@ func TestDeserializePayloadProtobuf(t *testing.T) {
...
+	}
+
+	sc := saramaConsumer{}
+	var messageHeaders = sc.convertToMessageHeaders(consumerHdrs)
+	require.Equal(t, len(messageHeaders), len(consumerHdrs))
+	for i, element := range messageHeaders {
+		require.Equal(t, string(consumerHdrs[i].Value), string(element.Value))
+		require.Equal(t, string(consumerHdrs[i].Key), string(element.Key))
+		//fmt.Println(string(consumerHdrs[i].Value))
+		i++
```
Comment (variadico) [inline server/kafka/consumer_test.go:141]: Looks like a left over comment. And also another unused `i++`.

**Comment by variadico** [comment]
Overall, looks good, but there's just a little bit of clean up to do.
```
for i, elem := range slice {
  // i automatically increases from 0 to len(slice)-1
  // no need to say i++
  // that increments a copy of the index, has no effect on the behavior of the loop
}
```

**Comment by rpuppo** [comment]
Hello @variadico , retrofits done, could you please re-validate?
BR,
Riccardo

**Review by variadico — APPROVED** [review]

**Comment by variadico** [comment]
Thank you very much, @rpuppo! 🎉

**Pull request merged**